src/client/dfs/dfs.c

```diff
@@ -257,18 +257,15 @@ static int
 fetch_entry(daos_handle_t oh, daos_handle_t th, const char *name,
 	    bool fetch_sym, bool *exists, struct dfs_entry *entry)
 {
-	daos_sg_list_t	sgls[INODE_AKEYS];
-	daos_iov_t	sg_iovs[INODE_AKEYS];
-	daos_iod_t	iods[INODE_AKEYS];
+	daos_sg_list_t	sgls[INODE_AKEYS + 1];
+	daos_iov_t	sg_iovs[INODE_AKEYS + 1];
+	daos_iod_t	iods[INODE_AKEYS + 1];
 	char		value[DFS_MAX_PATH];
 	daos_key_t	dkey;
 	unsigned int	akeys_nr, i;
 	int		rc;
 
-	if (name == NULL)
-		return -DER_INVAL;
-
-	D_DEBUG(DB_TRACE, "fetch entry %s\n", name);
+	D_ASSERT(name);
```

Comment:
On insert, you are inserting an oid akey for a normal object and syml akey for but never both.   Is it true the value will be "bogus" on fetch or non-existent?   How does fetch handle it when some values exist and others don't?   I guess if this is working, it must not be returning an error or it would fail every time.   Does the return have any indication that the value isn't actually returned or is it entirely up to the user to make sure they never read a syml or oid if it's not expected to be valid?

Reply:
so yes on insert, with this patch, you insert one or the other.
On fetch, you fetch both. but you also fetch the type and DFS checks the type and uses the oid or value accordingly. But yes on fetch it should be non-existent meaning the fetch just doesn't fill any of the return buffer. it doesn't return an error.